x-pack/platform/plugins/shared/alerting/server/application/maintenance_window/methods/create/create_maintenance_window.ts

```diff
@@ -65,7 +65,13 @@ export async function createMaintenanceWindow(
   }
 
   const id = SavedObjectsUtils.generateId();
-  const expirationDate = moment().utc().add(1, 'year').toISOString();
+  let expirationDate;
+  if (rRule.interval || rRule.freq) {
+    expirationDate = moment().utc().add(1, 'year').toISOString();
+  } else {
+    expirationDate = moment(rRule.dtstart).utc().add(duration, 'ms').toISOString();
```

Comment:
if maintenance window has non recurring schedule, we don't need to set expiration date for more +1 year.